javascript/ql/test/library-tests/CustomLoadStoreSteps/test.ql

```diff
@@ -0,0 +1,22 @@
+import javascript
+
+class Configuration extends TaintTracking::Configuration {
+  Configuration() { this = "PromiseFlowTestingConfig" }
+
+  override predicate isSource(DataFlow::Node source) {
+    source.getEnclosingExpr().getStringValue() = "source"
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    any(DataFlow::InvokeNode call | call.getCalleeName() = "sink").getAnArgument() = sink
+  }
+
+  // When the source code states that "foo" is being read, "bar" is additionally being read.
+  override predicate isAdditionalLoadStep(DataFlow::Node pred, DataFlow::Node succ, string prop) {
+    pred.(DataFlow::SourceNode).getAPropertyRead("foo") = succ and prop = "bar"
```

Comment:
I know this is just a test case, but it's generally a bad idea to use `getALocalSource` to obtain the predecessor of a step, which here happens through `getAPropertyRead`.